api/src/main/java/jakarta/persistence/NamedQuery.java

```diff
@@ -86,9 +87,34 @@
      * is specified, the query must be executed in a transaction
      * and the persistence context joined to the transaction.
      * @since 2.0
+     * @see Query#setLockMode
      */
     LockModeType lockMode() default LockModeType.NONE;
-    
+
+    /**
+     * (Optional) The {@linkplain CacheStoreMode cache store mode}
+     * to use in query execution.
+     * @since 4.0
+     * @see Query#setCacheStoreMode 
+     */
+    CacheStoreMode cacheStoreMode() default CacheStoreMode.USE;
+
+    /**
+     * (Optional) The {@linkplain CacheRetrieveMode cache retrieve mode}
+     * to use in query execution.
+     * @since 4.0
+     * @see Query#setCacheRetrieveMode 
+     */
+    CacheRetrieveMode cacheRetrieveMode() default CacheRetrieveMode.USE;
```

Comment:
I just though of a problem.  The default values  here will override any corresponding value that was set on the EntityManager, potentially causing a breaking change in behavior.  To do this without a breaking change, we would need a different default representing UNSPECIFIED.

Reply:
> we would need a different default representing `UNSPECIFIED`

Damn, I can't really think of a better option than this one. And I don't like it much, because setting it onto an `EntityManager` would have to be an error.

Maybe we should just put this issue on ice.